pagerduty/resource_pagerduty_team_membership.go

```diff
@@ -277,7 +277,7 @@ func dissociateEPsFromTeam(c *pagerduty.Client, teamID string, eps []string) ([]
 	for _, ep := range eps {
 		retryErr := retry.Retry(2*time.Minute, func() *retry.RetryError {
 			_, err := c.Teams.RemoveEscalationPolicy(teamID, ep)
-			if err != nil && !isErrCode(err, 404) {
+			if err != nil {
```

Comment:
Nice catch! But I think the original intention of `&& !isErrCode(err, 404)` is to skip retry times. Thus, we should return a `retry.RetryableError(err)` if `isErrCode(err, 404)` inside of this error handling.

Reply:
You certain it should not lead to calling continue at line 292? I think the 404 is exactly for that escalation policy not found - hence skipping it.
The current bug I experience in my environment is that since it is not calling continue - it adds the non relevant/existent escalation policy to the epsDissociatedFromTeam in line 295 and than breaks when it tried to add it back.

Reply:
I'm sorry, I meant `NonRetryableError`.
You're are right, and we're intending the same. Now (before your changes) the subfunction passed to retry.Retry returns `nil` when `err` is a 404. With your changes it will sleep and return a retryable error. I want to send the execution directly into 292 without waiting 2 minutes.

```
| case          | err == nil | err is 404                   | err is anything else      |
| ------------- | ---------- | ---------------------------- | ------------------------- |
| now           | nil        | nil                          | retry.RetryableError(err) |
| your changes  | nil        | retry.RetryableError(err)    | retry.RetryableError(err) |
| my suggestion | nil        | retry.NonRetryableError(err) | retry.RetryableError(err) |
```

Reply:
Ahh, make sense, adjusted the solution accordingly.